api/client/credentials.go

```diff
@@ -177,6 +180,92 @@ func (c *IdentityCreds) load() error {
 	return nil
 }
 
+// LoadProfile is used to load credentials from a tsh Profile.
+// If dir is not specified, the default profile path will be used.
+// If name is not specified, the current profile name will be used.
+func LoadProfile(dir, name string) *ProfileCreds {
+	return &ProfileCreds{
+		dir:  dir,
+		name: name,
+	}
+}
+
+// ProfileCreds are used to authenticate the client
+// with a tsh profile with the given directory and name.
+type ProfileCreds struct {
+	dir     string
+	name    string
+	profile *Profile
+}
+
+// Dialer is used to dial a connection to Auth.
+func (c *ProfileCreds) Dialer(keepAliveInterval, dialTimeout time.Duration) (ContextDialer, error) {
+	profile, err := ProfileFromDir(c.dir, c.name)
+	if err != nil {
+		return nil, trace.BadParameter("profile could not be decoded: %v", err)
+	}
+
+	sshConfig, err := profile.SSHClientConfig()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	dialer := NewTunnelDialer(*sshConfig, keepAliveInterval, dialTimeout)
+	return ContextDialerFunc(func(ctx context.Context, network, _ string) (conn net.Conn, err error) {
+		// Ping web proxy to retrieve tunnel proxy address.
+		pr, err := Find(ctx, profile.WebProxyAddr, false, nil)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		conn, err = dialer.DialContext(ctx, network, pr.Proxy.SSH.TunnelPublicAddr)
+		if err == nil {
+			return conn, nil
+		}
+		// not wrapping on purpose to preserve the original error
+		return nil, err
+	}), nil
+}
+
+// TLSConfig returns TLS configuration used to connect to Auth.
+func (c *ProfileCreds) TLSConfig() (*tls.Config, error) {
+	if err := c.load(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	tlsConfig, err := c.profile.TLSConfig()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return configure(tlsConfig), nil
+}
+
+// SSHClientConfig returns SSH configuration used to connect to Proxy.
+func (c *ProfileCreds) SSHClientConfig() (*ssh.ClientConfig, error) {
+	if err := c.load(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	sshConfig, err := c.profile.SSHClientConfig()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return sshConfig, nil
+}
+
+func (c *ProfileCreds) load() error {
+	if c.profile != nil {
+		return nil
+	}
+	var err error
+	if c.profile, err = ProfileFromDir(c.dir, c.name); err != nil {
+		return trace.BadParameter("profile could not be decoded: %v", err)
+	}
+	return nil
+}
```

Comment:
Mainly to catch the error returned by `load`. The loader methods like `client.LoadProfile` don't return an error so that they can be easily loaded into `client.Config`:

```go
clt, err := client.New(client.Config{
    Credentials: []client.Credentials{
        client.LoadProfile("", ""),
    }
})
```